fiftyone/factory/repos/execution_store.py

```diff
@@ -41,6 +41,10 @@ def __init__(self, dataset_id: Optional[ObjectId] = None, is_cache=False):
             dataset_id (Optional[ObjectId]): the dataset ID to operate on
             is_cache (False): whether the store is a cache store
         """
+        if dataset_id is not None and not isinstance(dataset_id, ObjectId):
```

Comment:
you can just do
```suggestion
        if dataset_id is not None:
            self._dataset_id = ObjectId(dataset_id)
```

This will auto convert the string to an ObjectId or raise an error otherwise

Reply:
Right, but for sanity I want the caller to handle that. I want to avoid having to support a string value at all.